src/Data/HashMap.js

```diff
@@ -477,6 +477,37 @@ MapNode.prototype.itraverse = function (pure, apply, f) {
     return m;
 }
 
+MapNode.prototype.any = function (predicate, shift) {
+  var dataMap = this.datamap;
+  var nodeMap = this.nodemap;
+
+  while (dataMap !== 0) {
+      var bit = lowestBit(dataMap);
+      var dataIndex = index(this.datamap, bit);
+      var value = this.content[dataIndex * 2 + 1];
+
+      if (predicate(value)) {
+          return true;
+      }
+
+      dataMap &= ~bit;
+  }
+
+  while (nodeMap !== 0) {
+      var bit = lowestBit(nodeMap);
+      var nodeIndex = index(this.nodemap, bit);
+      var subNode = this.content[this.content.length - nodeIndex - 1];
+
+      if (subNode.any(predicate, shift + 5)) {
+          return true;
+      }
+
+      nodeMap &= ~bit;
+  }
```

Comment:
IIRC (it's been a while), `this.content` has all the keys and values, then all the subnodes. Something like this for x key value pairs and y subnodes:
`[k1, v1, k2, v2, ... kx, vx, n1, n2, ..., ny]`
You should be able to get away without doing all that bitwise stuff in the loops. You only need to know how many key value pairs there are. Then you go through the contents array in steps of two for the data and at the right time switch to steps of one for the recursive calls.
Have a look at how `MapNode.prototye.ifoldMap` does it. (You don't need the keys for `any`, but you'd need them for `anyWithKey`).

Reply:
> and at the right time switch to steps of one for the recursive calls

How do i know if it's the right time?

Reply:
After 2*popcount(datamap) array elements, see `ifoldMap`.